torch/nested/_internal/ops.py

```diff
@@ -439,32 +439,93 @@ def linear_backward_default(func, *args, **kwargs):
     return (ds, dw, db)
 
 
-@register_jagged_func(torch.ops.aten._to_copy.default, "self: jt_all")
-def to_copy_default(func, *args, **kwargs):
-    from .nested_tensor import _tensor_symint_registry
-
+@register_jagged_func(
+    torch.ops.aten._to_copy.default,
+    "self: jt_all, dtype: any?, layout: any?, device: any?, pin_memory: any?, non_blocking: any?, memory_format: any?",
+)
+def _to_copy_default(func, *args, **kwargs):
     _, new_kwargs = normalize_function(
         func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
     )
 
-    inp = new_kwargs.pop("input")
-    # don't change layout
-    new_kwargs.pop("layout")
+    inp: NestedTensor = new_kwargs.pop("input")
+    new_layout = new_kwargs.pop("layout")
+    if new_layout is None:
+        new_layout = inp.layout
+
+    if new_layout not in [torch.strided, torch.jagged]:
+        raise ValueError("Nested Tensors can only have jagged and strided layouts")
 
     new_values = func(inp._values, **new_kwargs)
+
+    # Copy to a new Python subclass NestedTensor
     new_offsets = inp._offsets.to(device=new_values.device)
     _tensor_symint_registry[new_offsets] = _tensor_symint_registry[inp._offsets]
     inp_kwargs = extract_kwargs(inp)
     inp_kwargs["offsets"] = new_offsets
 
-    return NestedTensor(new_values, **inp_kwargs)
+    new_njt = NestedTensor(new_values, **inp_kwargs)
+
+    if new_layout == torch.jagged:
+        return new_njt
+
+    return torch._nested_jagged_to_strided(new_njt)
+
+
+@register_jagged_func(torch.ops.aten._nested_jagged_to_strided.default, "self: jt_all")
+def _nested_jagged_to_strided(func, *args, **kwargs):
+    _, new_kwargs = normalize_function(
+        func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
+    )
+
+    inp: NestedTensor = new_kwargs.pop("input")
+
+    # TODO: Figure out a better way to accomplish this?
+    if torch._subclasses.fake_tensor.is_fake(inp):
+        # NB: NST is not supported in PT2. Calling this op with garbage will hit the
+        # fake tensor unsupported impl and graph break.
+        return torch._nested_view_from_buffer(
+            inp._values.view(-1),
+            nested_size=inp._values,
+            nested_strides=inp._values,
+            offsets=inp._values,
+        )
```

Comment:
cc @bdhirsh in case you have a suggestion for the right way to handle this. The current implementation of this op doesn't work for both real and fake tensors, so this hack gets around this.

At a high level, we just want to graph break when this op is encountered during tracing since NSTs are not supported within PT2.

Reply:
oh - if you want to graph break here, then one thing that might work is you can detect if the input is fake (like you're already doing), and raise a `UnsupportedFakeTensorException`. See example here: https://github.com/pytorch/pytorch/blob/main/torch/_subclasses/fake_tensor.py#L325

Dynamo should catch that exception when it is running fake tensor prop and know to graph break